Design.md

```diff
@@ -213,7 +213,10 @@ This API will return the user.
 
 ### Get User with Canonical ID
 This API will return the user.
-1. The `get-user` API will be called using assumed role credentials with canonical_id as username
+1. The `get-account` API will be called using vaultclient with the provided `canonical-id` and account details will be used to fill tenant details of the response.
+1. The `list-users` API will be called with `offset` as 0 and `limit` as 1000 and the last user values in the response will be used to fill the user details of the response.
```

Comment:
We are giving back randomly any user back. The only thing I am afraid of is the mismatch of users in cloud director, it handles its own RBAC as well.

Reply:
Ceph implementation has a canonical ID in the format of `<Tenant_ID>/<USER_ID>` . Expectations were based on this format. As I have already mentioned in the subsequent bullet points, we cannot retrieve the specific user with respect to the canonicalID. Few important things to note here are:
* `get-user-with-canonical-id` API will be called by OSE **only** when user APIs does not return the `canonical-id` for any user ( `listusers`,`getuser`,`queryusers`)
* Probably this api might be called if the latest user did not return canonical id during `createuser` (which is not possible with current OSIS implementation)

Reply:
Okay. We should keep this in mind when troubleshooting any future issues.